### PR TITLE
DM-10125: Initial version

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: acronyms.tex [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+\textbf{Acronym} & \textbf{Description}  \\\hline
+API&Application Programming Interface \\\hline
+CAM&CAMera \\\hline
+CU&Coordination Unit (in DPAC) \\\hline
+DM&Data Management \\\hline
+DMLT&DM Leadership Team \\\hline
+DPAC&Data Processing and Analysis Consortium \\\hline
+DPC&Data Processing Centre \\\hline
+ESA&European Space Agency \\\hline
+HSC&Hyper Supreme-Cam \\\hline
```
Comment (timj) [inline acronyms.tex]: "Suprime"

File: acronyms.tex [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+LSST&Large-aperture Synoptic Survey Telescope \\\hline
+NCSA&National Center for Supercomputing Applications \\\hline
+OPS&OPerationS \\\hline
+SP&Software Product \\\hline
+SPR&Software Problem Report \\\hline
```
Comment (timj) [inline acronyms.tex]: Are SPRs JIRA tickets?

File: constraints.tex [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+There are requirements in \citeds{LSE-61} (aka \DMSR) which captures the \citeds{LSE-30} (\OSS) requirements that DM is responsible for. \textit{In practice \citeds{LSE-63} (the QA document) has not been flown down to \citeds{LSE-61}}. These are:
+
+\begin{itemize}
+
+\item The primary computational performance flown down from \citeds{LSE-29} (\LSR) is OTT1 which is the requirement to issue an alert within 60 seconds of exposure end.\dmreq{0004}\lsrreq{0101}
+
+\item Another requirement flows down from \citeds{LSE-29} is calculation of orbits within 24 hours of the end of the observing night\dmreq{0004}\lsrreq{0104}\reqparam{L1PublicT}
+
+\item There is a new (not yet baselined?) requirement for the calibration pipeline to reduce calibration observations within 1200 seconds
```
Comment (timj) [inline constraints.tex]: DMS-REQ-0131 and `calProcTime`.

File: constraints.tex [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+\item A nightly report on data quality, data management system performance and a calibration report have to be generated with 4 hours of the end of the night\dmreq{0096}\reqparam{dq\-Report\-Compl\-Time}
+
+\end{itemize}
+
+Note that there are no computational requirements on individual technical components eg. data processing cluster availability, database data retrieval speeds, etc. There is an upper limit on acceptable data loss, and there is a network availability requirement.
```
Comment (timj) [inline constraints.tex]: Some of the performance metrics are used as assumptions in the baselined sizing model LSE-81/LSE-82 but need to be materialized in DMSR (or component requirements documents).

File: body.tex [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+\input{tools}
+
+% ------------------------------- UNIT & INTEGRATION TESTS -------------------
+
+\section{Unit and Integration Tests}
+
+\subsection{Approach}
+
+Unit and Integration Tests will be automatically executed through the JUnit test framework. The descriptions of the test below are extracted from the test cases code and documentation.The results of Unit and Integration Test to be included in the Sofwtare Test Report will be generated automatically from the output of the execution of the tests by JUnit. A script will be provided to perform thes processing steps.
```
Comment (timj) [inline body.tex]: JUnit? The developer guide has a whole section on unit testing.

File: body.tex [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+Module identification? (module tag in class header? mapping file?)
+
+\subsection{Test Coverage}
+
+Test coverage goal for unit and integration testing. Each class and public method shall have a JUnit test harness that may be labelled according to their purpose (e.g. I/O, individual class/method tests, software integration, data model integration etc.). Nominal and contingency
+tests should be clearly identified.
+
+Interface coverage...
+
+The tool [insert name of unit test coverage tool here] will be used to provide metrics on the code coverage by Unit Tests for \product \ and this metric will be provide in the Test Report.
```
Comment (timj) [inline body.tex]: we have a plan for adding test coverage to the builds via pytest plus coverage.py. I don't think work has been scheduled for it though.

File: body.tex [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+\subsection{Approach}
+
+Unit and Integration Tests will be automatically executed through the JUnit test framework. The descriptions of the test below are extracted from the test cases code and documentation.The results of Unit and Integration Test to be included in the Sofwtare Test Report will be generated automatically from the output of the execution of the tests by JUnit. A script will be provided to perform thes processing steps.
+
+Module identification? (module tag in class header? mapping file?)
+
+\subsection{Test Coverage}
+
+Test coverage goal for unit and integration testing. Each class and public method shall have a JUnit test harness that may be labelled according to their purpose (e.g. I/O, individual class/method tests, software integration, data model integration etc.). Nominal and contingency
```
Comment (timj) [inline body.tex]: What does this mean for the LSST code base?

File: body.tex (hunk 6 above)
Comment (timj) [inline body.tex]: Some of this QA is also covered in LDM-522.

File: validation.tex [hunk 8]
```diff
@@ -0,0 +1,86 @@
+
+\section{Validation Tests}
+ Validation of the system though Oerations Rehersals (and or end to end tests)
```
Comment (timj) [inline validation.tex]: Data Challenges as well?

File: validation.tex [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+\subsection{General strategy}
+
+Description of the general verification and validation strategy, decomposition into verification testing categories (e.g. science tests, SP external interface tests, algorithms interrelation and sequence). Assessed validation tests results shall be available over the software development duration: they are stored into SVN repository along with related input data, property-file, etc.
+
+A subset of tests are run at DPC during software release qualification process, the results of DPC runs are compared with corresponding test outputs. During DPC integration tests, these assessed outputs will also allow to verify software non-regression.
```
Comment (timj) [inline validation.tex]: Are we using the term DPC now?

File: validation.tex [hunk 10]
```diff
@@ -0,0 +1,86 @@
+
+\section{Validation Tests}
+ Validation of the system though Oerations Rehersals (and or end to end tests)
+
+\subsection{General strategy}
+
+Description of the general verification and validation strategy, decomposition into verification testing categories (e.g. science tests, SP external interface tests, algorithms interrelation and sequence). Assessed validation tests results shall be available over the software development duration: they are stored into SVN repository along with related input data, property-file, etc.
```
Comment (timj) [inline validation.tex]: Not SVN (I hope). We don't generally use "SP" as a term yet.

File: validation.tex [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+A subset of tests are run at DPC during software release qualification process, the results of DPC runs are compared with corresponding test outputs. During DPC integration tests, these assessed outputs will also allow to verify software non-regression.
+ 
+\subsection{Test Designs}
+
+\subsubsection{Test Design \CU-\product-SYS-X}
+
+\paragraph{Objective}
```
Comment (timj) [inline validation.tex]: Would these be JIRA tickets or prose in a PDF?

File: scivalidation.tex [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+tests and data analyses to exercise the integrated system and assess its
+quality relative to expectations for the current phase of construction. 
+This assessment is fed back to DM Subsystem Science and Systems Engineering
+teams to inform them about the status and needed improvements to the system.
+
+Beyond reporting on the results, the SV team examines the tests or
+procedures developed in this phase and identifies those that are good new
+metrics of system quality and could be run in an automated fashon.  These
+are fed back to the development teams for productizing and incorporation
+into the automated QC systems.
```
Comment (timj) [inline scivalidation.tex]: QC is not defined in this document.

File: scivalidation.tex [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+to make it more consistent with the system vision), as well as the overall
+system vision (i.e., recommendations on where the vision may need to be
+modified to fully respond to stakeholder needs).
+
+\item Measurements of performance metrics that do not lend themselves to
+easy automation (e.g., science activities requiring human involvement, like
+visual classification, or UX tests).
+
+\item Identification of new performance metrics to be tracked, including
+potential deliveries of code to the DM Construction and I\&T teams for
```
Comment (timj) [inline scivalidation.tex]: Define I&T somewhere.

File: scivalidation.tex [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+\caption{Organogram of the Data Management Science Validation Team. 
+The group is chaired by the DM Science Validation Scientist,
+with the DM Science Pipelines Scientist and Institutional Science Leads making up
+the permanent membership. Depending on the SV activities being executed at any
+given time, the group may draw on additional temporary members from DM SST Staff,
+the broader DM Construction staff, as well as external scientists (e.g.,
+Science Collaboration members committed to assisting SV goals). SV membership
+is reassessed on a cycle by cycle basis, with estimates incorporated in the
+long-term plan.
+\label{fig:DMsvg}}
```
Comment (timj) [inline scivalidation.tex]: Figure is never referenced from the text. I think the bulk of this caption should be integrated into the main body of text (and much of it already is). What do the colors of the boxes mean?

File: scivalidation.tex [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+successful execution of DM Science Validation activities.  This
+responsibility is delegated to the {\bf DM Science Validation Scientist},
+who leads the Science Validation (SV) team. 
+
+The SV team guides the definition of goals and receives the products of
+dress rehearsal activities, consistent with the long-term testing roadmap
+defined in Section~\ref{sect:schedule}.  Decisions on strategic goals of SV exercises are made
+in close consultation and coordination with the DM Project Manager and
+Subsystem Scientist.  The results of SV activities are reported to the DM
+Project Manager and Subsystem Scientist.
```
Comment (timj) [inline scivalidation.tex]: Duplication of "DM Project Manager and Subsystem Scientist". Maybe end previous sentence with ", who will also receive a report on the results of SV activities."?

File: scivalidation.tex [hunk 16]
```diff
@@ -0,0 +1,193 @@
...
+Managers are not formal members, but monitor the work of the group.
+
+\subsubsection{Example}
+
+An example of a Science Validation activity may be as follows:
+
+\begin{itemize}
+
+\item Based on the long-term development roadmap and new capabilities
+expected to be delivered, the at the beginning of a 6-month cycle the SV
```
Comment (timj) [inline scivalidation.tex]: "the at the"

File: acronyms.tex (hunk 1 above)
Comment (womullan) [inline acronyms.tex]: you should not edit this file .. edit myacronyms.tex

File: acronyms.tex (hunk 2 above)
Comment (womullan) [inline acronyms.tex]: yes for LSST

File: body.tex (hunk 7 above)
Comment (womullan) [inline body.tex]: We should decide - I guess you do not categorize your tests ? We named ourse like SciTestName and PerfTestName so I coulrun science tests (which may take a long time) seperately

File: validation.tex (hunk 8 above)
Comment (womullan) [inline validation.tex]: yes - somewhere it included data challenges

File: validation.tex (hunk 9 above)
Comment (womullan) [inline validation.tex]: I just dumped some text .. not carefully parsed .. but we could use DPC :)

File: validation.tex (hunk 11 above)
Comment (womullan) [inline validation.tex]: Either ..

**Comment by timj** [comment]
Does this mean that you have approved LDM-503 as a released version and I should upload it to DocuShare?

**Comment by womullan** [comment]
no .. I merged your pull request to my ticket branch right .. ??

**Comment by timj** [comment]
Unfortunately I was commenting on your ticket branch, not making edits. I backed out this merge and created a new PR on #3.

**Pull request merged**